### PR TITLE
Add tryout UMA doc

### PR DESCRIPTION
**Pull request by hwupathum**
Add User Managed doc from https://is.docs.wso2.com/en/latest/guides/access-delegation/uma/. Since the connector is removed from the IS, it is removed from the docs as well.

### Related Issues
- https://github.com/wso2/product-is/issues/15093

### CODE REVIEW

File: docs/tryout.md [hunk 1]
```diff
@@ -0,0 +1,390 @@
+# User Managed Access with WSO2 Identity Server
+
+WSO2 Identity Server (WSO2 IS) supports the UMA 2.0 protocol, which allows a resource owner to easily share resources with other requesting parties. This tutorial demonstrates how you can use WSO2 IS as the authorization server to try out UMA for a sample scenario.
+
+## Introduction
+
+To demonstrate the UMA flow using WSO2 IS as the authorization server, let’s consider a sample scenario where the resource owner wants to share a resource on their resource server with a requesting party. This requesting party uses a client to view the resource owner's resource. Note that the resource owner wants to allow this requesting party to only have view permission to their resource.
+
+Once you try out the guide you will understand how to use UMA 2.0 to do the following:
```
Comment (nilminiwso2) [inline docs/tryout.md:9]:
```suggestion
Once you try out the guide, you will understand how to use UMA 2.0 to do the following:
```

File: docs/tryout.md [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+    single console (authorization server).
+
+---
+
+## Create the resource owner
+
+
+1.  Log in to the WSO2 Identity Server Management Console (`https://<IS_HOST>:<PORT>/carbon`) using administrator credentials (`admin:admin`).
+2.  On the **Main** > **Identity** section, click **Add** under **Users and Roles**.
+<a name ="resourceowner_credentials"></a>
```
Comment (nilminiwso2) [inline docs/tryout.md:22]: is this a link?

File: docs/tryout.md [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+3. Click **Add New User** and create a new user by providing username and password.
```
Comment (nilminiwso2) [inline docs/tryout.md:23]:
```suggestion
3. Click **Add New User** and create a new user by providing the username and password.
```

File: docs/tryout.md [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+4.  Click **Next**.
+5.  Select the role as **admin**.
+6.  Click **Finish**.
+
+---
+
+## Create the requesting party
+
+1.  On the **Main** > **Identity** section, click **Add** under **Users and Roles**.
+2.  Click **Add New User** and create a new user by providing username and password.
```
Comment (nilminiwso2) [inline docs/tryout.md:33]:
```suggestion
2.  Click **Add New User** and create a new user by providing the username and password.
```

File: docs/tryout.md [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+3.  Click **Finish**.
+
+Now that you have the resource owner and requesting party to try out the scenario, next step is to configure one service provider for the resource server and another service provider for the client acting on behalf of the requesting party.
```
Comment (nilminiwso2) [inline docs/tryout.md:36]:
```suggestion
Now that you have the resource owner and requesting party to try out the scenario, the next step is to configure one service provider for the resource server and another service provider for the client acting on behalf of the requesting party.
```

File: docs/tryout.md [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+
+---
+
+## Configure service provider to act as the resource server
+
+You need to register your application as a service provider in WSO2 Identity Server.
+
+1. Log in to the WSO2 Identity Server Management Console (`https://<IS_HOST>:<PORT>/carbon`) using administrator credentials (`admin:admin`).
+
+2. Navigate to **Main** > **Identity** > **Service Providers** > **Add**.
```
Comment (nilminiwso2) [inline docs/tryout.md:46]:
```suggestion
2. Go to **Main** > **Identity** > **Service Providers** and click **Add**.
```

File: docs/tryout.md [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+
+3. Enter a **Service Provider Name**. Optionally, enter a **Description**.
+    
+4. Click **Register**.    
+
+### Basic OAuth/OpenID Connect configuration
+
+1. Expand **Inbound Authentication Configuration** and then **OAuth/OpenID Connect Configuration**. 
+
+2. Click **Configure.**
```
Comment (nilminiwso2) [inline docs/tryout.md:56]:
```suggestion
2. Click **Configure**.   
```

File: docs/tryout.md [hunk 8]
```diff
@@ -0,0 +1,390 @@
...
+    
+---
+    
+## Configure service provider to act as the client
+    
+You need to register your application as a service provider in WSO2 Identity Server.
+
+1. Log in to the WSO2 Identity Server Management Console (`https://<IS_HOST>:<PORT>/carbon`) using administrator credentials (`admin:admin`).
+
+2. Navigate to **Main** > **Identity** > **Service Providers** > **Add**.
```
Comment (nilminiwso2) [inline docs/tryout.md:74]:
```suggestion
2. Go to **Main** > **Identity** > **Service Providers** and click **Add**.
```

File: docs/tryout.md [hunk 9]
```diff
@@ -0,0 +1,390 @@
...
+
+1. Expand **Inbound Authentication Configuration** > **OAuth/OpenID Connect Configuration** and click **Configure**.
+
+2. Enter the **Callback Url**. 
+
+3. Click **Add**. Note the **OAuth Client Key** and **OAuth Client Secret** that appear. 
+
+### Claim configuration
+
+Update claims for service provider:
```
Comment (nilminiwso2) [inline docs/tryout.md:92]:
```suggestion
Update claims for the service provider:
```

File: docs/tryout.md [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+    ```
+
+---
+
+## Register the resource
+
+Now, you need to register the resource.
+
+-   Execute the following curl command to put the resouce owner's resource under authorization server (WSO2 IS) protection.
+    -   Make sure to replace the `<PAT>` tag with the [access token you got in the previous section](#obtain-the-protection-api-access-token-pat).
```
Comment (nilminiwso2) [inline docs/tryout.md:139]:
```suggestion
    -   Be sure to replace the `<PAT>` tag with the [access token you got in the previous section](#obtain-the-protection-api-access-token-pat).
```

Comment (nilminiwso2) [reply docs/tryout.md:139]: Can we make this a note?

File: docs/tryout.md [hunk 11]
```diff
@@ -0,0 +1,390 @@
...
+
+    ```
+    {
+        "_id": "ceaa6506-1da9-456b-88d8-027797d2e081"
+    }
+    ```
+
+Now you have completed registering the resource.
+
+Next, you need to create and publish an access policy to provide specific users appropriate resource scopes (permissions) to access the resource. <br> e.g. The requesting party is allowed to have view permission to the registered resource of the resource owner.
```
Comment (nilminiwso2) [inline docs/tryout.md:179]:
```suggestion
Next, you need to create and publish an access policy to provide specific users appropriate resource scopes (permissions) to access the resource. For example, the requesting party is allowed to have view permission to the registered resource of the resource owner.
```

File: docs/tryout.md [hunk 12]
```diff
@@ -0,0 +1,390 @@
...
+
+---
+
+## Publish a policy
+
+Follow the steps given below to create, register and publish a policy:
```
Comment (nilminiwso2) [inline docs/tryout.md:185]:
```suggestion
Follow the steps given below to create, register, and publish a policy:
```

File: docs/tryout.md [hunk 13]
```diff
@@ -0,0 +1,390 @@
...
+
+1.  Sign in to the management console (`https://<IS_HOST>:<IS_PORT>/carbon`) using resource owner [credentials](#create-the-resource-owner).
+2.  On the **Main** tab, go to the **Entitlement** section and click **Policy Administration** under **PAP**.
+3.  Click **Add New Entitlement Policy** and then click **Write Policy in XML**.
+4.  Copy the following sample policy and paste it on the **Source View** pane:
+
+- Replace the `{ENTER_YOUR_RESOURCE_ID}` tag with the resource ID that you obtained when you [registered the resource](#register-the-resource).
```
Comment (nilminiwso2) [inline docs/tryout.md:192]: this list should be formatted under number 4 afaiu.

File: docs/tryout.md (hunk 2 above)
Comment (hwupathum) [reply docs/tryout.md:22]: No its not

File: docs/tryout.md (hunk 10 above)
Comment (hwupathum) [reply docs/tryout.md:139]: Is it possible to add notes to a github doc?

File: docs/tryout.md [hunk 14]
```diff
@@ -0,0 +1,390 @@
...
+                </Apply>
+                <AttributeDesignator AttributeId="http://wso2.org/identity/identity-action/action-name" Category="http://wso2.org/identity/identity-action" DataType="http://www.w3.org/2001/XMLSchema#string" MustBePresent="true"></AttributeDesignator>
+                </Apply>
+            </Condition>
+        </Rule>
+        <Rule Effect="Deny" RuleId="Deny_all"></Rule>
+    </Policy>
+    ```
+
+5.  **Save** the policy. You will see the new policy that you created listed in the **Policy Administration** page.
```
Comment (nilminiwso2) [inline docs/tryout.md:233]:
```suggestion
5.  Save the policy. You will see the new policy that you created listed in the **Policy Administration** page.  
```

File: docs/tryout.md [hunk 15]
```diff
@@ -0,0 +1,390 @@
...
+
+6.  Select the policy and click **Publish To My PDP**. This displays the **Publish Policy** page.
+
+7.  Click **Publish**. This displays a message for confirmation to publish the policy.
+8.  Click **Yes**. This publishes the policy. If you want to view the published policy, click **Policy View** under **PDP**. 
+     
+    ![view-uma-policy](img/view-uma-policy.png) 
+
+
+Now that the resource owner has put their resource under the authorization server protection,
```
Comment (nilminiwso2) [inline docs/tryout.md:245]:
```suggestion
Now that the resource owner has put their resource under the authorization server's protection, 
```

File: docs/tryout.md [hunk 16]
```diff
@@ -0,0 +1,390 @@
...
+    ```
+
+---
+
+## Obtain the OIDC id\_token
+
+The client should pass id token to prove its identity to the authorization server. For the sample scenario in this tutorial only the requesting party username is required.
+
+-   Execute the following curl command to obtain the OIDC id\_token.
+    -   Be sure to replace the `<CLIENT_ID>` and `<CLIENT_SECRET>` tags with the values you got when you [Configured the service provider for the client](#configure-service-provider-to-act-as-the-client).
```
Comment (nilminiwso2) [inline docs/tryout.md:282]:
```suggestion
    -   Be sure to replace the `<CLIENT_ID>` and `<CLIENT_SECRET>` tags with the values you got when you [configured the service provider for the client](#configure-service-provider-to-act-as-the-client).  
```

File: docs/tryout.md [hunk 17]
```diff
@@ -0,0 +1,390 @@
...
+
+---
+
+## Obtain the requesting party token
+
+The client acting on behalf of the requesting party has to obtain the requesting party token (RPT) with the obtained permission ticket and the claim token.
+
+Execute the following curl command to obtain the RPT.  
+
+-   Make sure to replace the `<CLIENT_ID>` and `<CLIENT_SECRET>` tags with the values you got after [configuring service provider for the client](#configure-service-provider-to-act-as-the-client).
```
Comment (nilminiwso2) [inline docs/tryout.md:316]:
```suggestion
-   Be sure to replace the `<CLIENT_ID>` and `<CLIENT_SECRET>` tags with the values you got after [configuring service provider for the client](#configure-service-provider-to-act-as-the-client).
```

File: docs/tryout.md [hunk 18]
```diff
@@ -0,0 +1,390 @@
...
+-   Replace `<PERMISSION_TICKET>` with the value you generated under the [obtaining a permission ticket](#obtain-a-permission-ticket) section.
```
Comment (nilminiwso2) [inline docs/tryout.md:317]:
```suggestion
-   Replace `<PERMISSION_TICKET>` with the value you generated when [obtaining a permission ticket](#obtain-a-permission-ticket).
```

File: docs/tryout.md [hunk 19]
```diff
@@ -0,0 +1,390 @@
...
+-   Make sure to replace the `<ID_TOKEN>` tag with the [OIDC id\_token](#obtain-the-oidc-id95token) you obtained.
```
Comment (nilminiwso2) [inline docs/tryout.md:318]:
```suggestion
-   Be sure to replace the `<ID_TOKEN>` tag with the [OIDC id\_token](#obtain-the-oidc-id95token) you obtained.
```

File: docs/tryout.md [hunk 20]
```diff
@@ -0,0 +1,390 @@
...
+    ```
+
+---
+
+## Obtain the requesting party token
+
+The client acting on behalf of the requesting party has to obtain the requesting party token (RPT) with the obtained permission ticket and the claim token.
+
+Execute the following curl command to obtain the RPT.  
+
```
Comment (nilminiwso2) [inline docs/tryout.md:315]: We can put the following three points into a note:

```
Be sure to replace the placeholders in the curl command as follows:

-  ..
- ...
- ...

```

File: docs/tryout.md (hunk 2 above)
Comment (nilminiwso2) [reply docs/tryout.md:22]: oh.. it renders as `admin:admin`. Nice.
Not sure how you got that to work.. :)

File: docs/tryout.md [hunk 21]
```diff
@@ -0,0 +1,390 @@
...
+-   Outsource authorization.
+-   Control resources and share resources in a selective manner via a
+    single console (authorization server).
+
+---
+
+## Create the resource owner
+
+
+1.  Log in to the WSO2 Identity Server Management Console (`https://<IS_HOST>:<PORT>/carbon`) using administrator credentials (`admin:admin`).
```
Comment (nilminiwso2) [inline docs/tryout.md:20]: Let's use `sign in` shall we?

File: docs/tryout.md [hunk 22]
```diff
@@ -0,0 +1,390 @@
...
+Once you try out the guide, you will understand how to use UMA 2.0 to do the following:
+
+-   Outsource authorization.
+-   Control resources and share resources in a selective manner via a
+    single console (authorization server).
+
+---
+
+## Create the resource owner
+
```
Comment (nilminiwso2) [inline docs/tryout.md:18]:
```suggestion
Let's create a resource owner in WSO2 IS.
```

File: docs/tryout.md [hunk 23]
```diff
@@ -0,0 +1,390 @@
...
+<a name ="resourceowner_credentials"></a>
+3. Click **Add New User** and create a new user by providing the username and password.
+4.  Click **Next**.
+5.  Select the role as **admin**.
+6.  Click **Finish**.
+
+---
+
+## Create the requesting party
+
```
Comment (nilminiwso2) [inline docs/tryout.md:31]:
```suggestion
Let's create the requesting party on WSO2 IS.
```

File: docs/tryout.md [hunk 24]
```diff
@@ -0,0 +1,390 @@
...
+1.  On the **Main** > **Identity** section, click **Add** under **Users and Roles**.
+2.  Click **Add New User** and create a new user by providing username and password.
+3.  Click **Finish**.
+
+Now that you have the resource owner and requesting party to try out the scenario, the next step is to configure one service provider for the resource server and another service provider for the client acting on behalf of the requesting party.
+
+---
+
+## Configure service provider to act as the resource server
```
Comment (nilminiwso2) [inline docs/tryout.md:40]: Shouldn't this be `Register application as a service provider`?

Comment (nilminiwso2) [reply docs/tryout.md:40]: Application is the resource server?
got a bit confused with the terminology

Comment (nilminiwso2) [reply docs/tryout.md:40]: Or we can say `Register the resource server as a service provider`

File: docs/tryout.md [hunk 25]
```diff
@@ -0,0 +1,390 @@
...
+
+You need to register your application as a service provider in WSO2 Identity Server.
```
Comment (nilminiwso2) [inline docs/tryout.md:42]:
```suggestion
You need to register your resource server (application) as a service provider in WSO2 Identity Server.
```

Comment (nilminiwso2) [reply docs/tryout.md:42]: will this work?

File: docs/tryout.md [hunk 26]
```diff
@@ -0,0 +1,390 @@
...
+
+    ![uma-grant-type](img/uma-grant-type.png)
+        
+4. Enter the **Callback Url**.
+
+5. Click **Add**. Note the **OAuth Client Key** and **OAuth Client Secret** that appear.
+    
+---
+    
+## Configure service provider to act as the client
```
Comment (nilminiwso2) [inline docs/tryout.md:68]: `Register the client as a service provider in WSO2 IS`?

File: docs/tryout.md [hunk 27]
```diff
@@ -0,0 +1,390 @@
...
+    
+You need to register your application as a service provider in WSO2 Identity Server.
```
[the remark this answers is not in the document]
Comment (nilminiwso2) [reply docs/tryout.md:70]: is this correct?
Since we are registering two applications, thought of adding more clarity

File: docs/tryout.md [hunk 28]
```diff
@@ -0,0 +1,390 @@
...
+
+1. Log in to the WSO2 Identity Server Management Console (`https://<IS_HOST>:<PORT>/carbon`) using administrator credentials (`admin:admin`).
```
Comment (nilminiwso2) [inline docs/tryout.md:72]: Sign in

File: docs/tryout.md [hunk 29]
```diff
@@ -0,0 +1,390 @@
...
+```
+
+---
+
+## Token introspection
+
+
+- Execute the following curl command to get the token introspection:  
+
+    -   Make sure to replace the `<PAT>` tag with the [access token you got in the previous section](#obtain-the-protection-api-access-token-pat).
```
Comment (nilminiwso2) [inline docs/tryout.md:341]:
```suggestion
    -   Be sure to replace the `<PAT>` tag with the [access token you got in the previous section](#obtain-the-protection-api-access-token-pat).
```

File: docs/tryout.md [hunk 30]
```diff
@@ -0,0 +1,390 @@
...
+    "exp":1553418559,
+    "iat":1553414959,
+    "client_id":"JfTSiJ24gh8sYHTQVuOl5RoftkAa",
+    "username":"sam"
+    }
+    ```
+
+-   If the token introspection for the RPT is successful, the resource
+server can share the resource with the client.
+    -   In order to obtain UMA related information in the introspection endpoint, add the following configuration to the `deployment.toml` file in the `<IS_HOME>/repository/conf/` folder.
```
Comment (nilminiwso2) [inline docs/tryout.md:373]:
```suggestion
    -   In order to obtain UMA-related information in the introspection endpoint, add the following configuration to the `deployment.toml` file in the `<IS_HOME>/repository/conf/` folder.  
```

File: docs/tryout.md [hunk 31]
```diff
@@ -0,0 +1,390 @@
...
+    "iat":1553414959,
+    "client_id":"JfTSiJ24gh8sYHTQVuOl5RoftkAa",
+    "username":"sam"
+    }
+    ```
+
+-   If the token introspection for the RPT is successful, the resource
+server can share the resource with the client.
+    -   In order to obtain UMA related information in the introspection endpoint, add the following configuration to the `deployment.toml` file in the `<IS_HOME>/repository/conf/` folder.  
+    -   This is disabled by default. The response shown above with additional UMA related details is what we get when the following configuration is enabled.
```
Comment (nilminiwso2) [inline docs/tryout.md:374]: I believe this point should be changed as given below and also moved to the last point.

```
This configuration is disabled by default. You will get the response shown above with additional UMA-related details when the configuration is enabled.
```

File: docs/tryout.md [hunk 32]
```diff
@@ -0,0 +1,390 @@
...
+To demonstrate the UMA flow using WSO2 IS as the authorization server, let’s consider a sample scenario where the resource owner wants to share a resource on their resource server with a requesting party. This requesting party uses a client to view the resource owner's resource. Note that the resource owner wants to allow this requesting party to only have view permission to their resource.
+
+Once you try out the guide, you will understand how to use UMA 2.0 to do the following:
+
+-   Outsource authorization.
+-   Control resources and share resources in a selective manner via a
+    single console (authorization server).
+
+---
+
```
Comment (nilminiwso2) [inline docs/tryout.md:16]: Should we add this?

```
## Prerequisites

Be sure to [enable UMA in your WSO2 Identity Server](https://github.com/wso2-extensions/identity-oauth-uma/blob/master/docs/config.md) before trying out this tutorial.

This UMA extension is compatible with **WSO2 IS 6.1.0** and later versions.
```

Comment (hwupathum) [reply docs/tryout.md:16]: These are added in the config.md

**Review by ashensw — APPROVED** [review]

**Pull request merged**